client/packages/system/src/Item/api/hooks/useItemVariant/useItemVariant.ts

```diff
@@ -0,0 +1,117 @@
+import { useEffect, useRef, useState } from 'react';
+import {
+  FnUtils,
+  isEmpty,
+  useMutation,
+  useTranslation,
+} from '@openmsupply-client/common';
+import { ItemVariantFragment } from '../../operations.generated';
+import { useItemApi, useItemGraphQL } from '../useItemApi';
+
+export function useItemVariant({
+  itemId,
+  variant,
+}: {
+  itemId: string;
+  variant: ItemVariantFragment | null;
+}) {
+  const t = useTranslation();
+  const { mutateAsync } = useUpsert({
+    itemId,
+  });
+
+  const [draft, setDraft] = useState<ItemVariantFragment>(
+    variant ?? {
+      __typename: 'ItemVariantNode',
+      id: FnUtils.generateUUID(),
+      name: '',
+      manufacturerId: '',
+      coldStorageTypeId: '',
+      packagingVariants: [
+        {
+          __typename: 'PackagingVariantNode',
+          id: '1',
+          packagingLevel: 1,
+          name: t('label.primary'),
+          packSize: 1,
+          volumePerUnit: 1,
+        },
+        {
+          __typename: 'PackagingVariantNode',
+          id: '2',
+          packagingLevel: 2,
+          name: t('label.secondary'),
+          packSize: 2,
+          volumePerUnit: 2,
+        },
+        {
+          __typename: 'PackagingVariantNode',
+          id: '3',
+          packagingLevel: 3,
+          name: t('label.tertiary'),
+          packSize: 3,
+          volumePerUnit: 3,
+        },
+      ],
+    }
+  );
+
+  const draftRef = useRef(draft);
+
+  useEffect(() => {
+    draftRef.current = draft;
+  }, [draft]);
```

Comment:
Not stoked with this, I don't think it should be required - there's something up with the packaging table, I think the `setter` is a closure over `updateDraft` that had stale draft state. Was a rabbit hole so just using a ref here to ensure packaging variant table updates don't clear the rest of the draft state, but would like to understand it more...